### ISSUE TITLE
CONTRIBUTING.rst: Recommend local testing offline?

### ISSUE THREAD

**Issue by MarkusPiotrowski** [biopython/biopython#3845]
I just get stuck by running the test suite online. As usual, `test_NCBI_qblast.py` took ages, so I stopped and started over, running the test suite offline.

I wonder if we should recommend offline testing as standard in our 'CONTRIBUTING.rst'. I fear that the recommended procedure (which will result in testing with online modules) will deter many new contributors (and irregular contributors like me) from running the test suite. 
I would suggest to change `CONTRIBUTING.rst` like so:

---------------

Local Testing
-------------

Please always run the style checks (see above) and the full test suite on your local computer before submitting a pull request, e.g.::

    $ git commit Bio/XXX.py Tests/test_XXX.py  -m "Fixed bug 123"
    $ python setup.py build
    $ python setup.py test --offline

If you have worked on a module, which requires internet access, test this without the ``--offline`` switch::

    $ python setup.py test Tests/test_XXX.py   # if XXX.py is an online module

If you have multiple versions of Python installed, ideally test them all (the Python tool ``tox`` can be helpful here). Ideally, run the full test suite also online.  

-------------

**Comment by peterjc** [comment]
I agree something should be changed there, but think rather we should cut down the online NCBI BLAST test to a bare minimum?

**Comment by MarkusPiotrowski** [comment]
It is already small, running 'only' 7 tests. And I'm not sure that one can guarantee that it's not already blocked when running the first test?
The tests will also run offline (because there are mocked versions included). So do you think we should remove online testing of this module from the regular test suite and only test this once in a while (e.g. when making a new release)?

**Comment by MarkusPiotrowski** [comment]
Or maybe we implement an optional `Timeout` argument in `Bio.Blast.NCBIWWW` which would (a) allow the user to stop waiting after a certain time and (b) could be set to a reasonable short time for testing?

**Comment by peterjc** [comment]
The timeout idea is attractive- how hard would it be though?

**Comment by MarkusPiotrowski** [comment]
Doesn't seem to be hard, `qblast` is running an endless `while` loop while waiting for a valid response (see below). One could add another timer and break out of the loop after a certain time, maybe raising a 'Timeout' Biopython error. `timeout` would be an optional keyword argument of `qblast`, defaulting to `None`.
`run_test.py` could be modified to handle this as a special case, if necessary. 

https://github.com/biopython/biopython/blob/6e4b0a97acd1b6beb99ffc0c67a95ede7adab887/Bio/Blast/NCBIWWW.py#L240-L273

**Comment by peterjc** [comment]
Adding a timeout option to NCBIWWW sounds good, then ``test_NCBI_qblast.py`` could set it explicitly - and if triggered abort the tests with an exception telling ``run_tests.py`` to treat it as skipped.
